lib/go/test/flow_qc_test.go

```diff
@@ -326,4 +429,193 @@ func TestQuorumCertificate(t *testing.T) {
+
+		signAndSubmit(
+			t, b, tx,
+			[]flow.Address{b.ServiceKey().Address, QCAddress},
+			[]crypto.Signer{b.ServiceKey().Signer(), QCSigner},
+			false,
+		)
+	})
+
+	numberOfClusters := 2
+	numberOfNodesPerCluster := 3
+
+	// Create new user accounts
+	addresses, _, signers := registerAndMintManyAccounts(t, b, accountKeys, numberOfClusters*numberOfNodesPerCluster)
+
+	clusterNodeIDStrings := make([][]string, numberOfClusters*numberOfNodesPerCluster)
+
+	stakingPrivateKeys, stakingPublicKeys, _, _ := generateManyNodeKeys(t, numberOfClusters*numberOfNodesPerCluster)
+
+	clusters := initClusters(clusterNodeIDStrings, numberOfClusters, numberOfNodesPerCluster)
+
+	t.Run("Should start voting with the admin with more nodes and clusters", func(t *testing.T) {
+
+		tx := createTxWithTemplateAndAuthorizer(b, templates.GenerateStartVotingScript(env), QCAddress)
+
+		err := tx.AddArgument(cadence.NewArray(clusters[0]))
+		require.NoError(t, err)
+
+		err = tx.AddArgument(cadence.NewArray(clusters[1]))
+		require.NoError(t, err)
+
+		err = tx.AddArgument(cadence.NewArray(clusters[2]))
+		require.NoError(t, err)
+
+		signAndSubmit(
+			t, b, tx,
+			[]flow.Address{b.ServiceKey().Address, QCAddress},
+			[]crypto.Signer{b.ServiceKey().Signer(), QCSigner},
+			false,
+		)
+
+	})
+
+	t.Run("Should claim voter resources for new accounts", func(t *testing.T) {
+
+		for i := 0; i < numberOfClusters*numberOfNodesPerCluster; i++ {
+
+			tx := createTxWithTemplateAndAuthorizer(b, templates.GenerateCreateVoterScript(env), addresses[i])
+
+			_ = tx.AddArgument(cadence.NewAddress(QCAddress))
+			_ = tx.AddArgument(cadence.NewString(clusterNodeIDStrings[i/3][i%3]))
+			_ = tx.AddArgument(cadence.NewString(stakingPublicKeys[i]))
+
+			signAndSubmit(
+				t, b, tx,
+				[]flow.Address{b.ServiceKey().Address, addresses[i]},
+				[]crypto.Signer{b.ServiceKey().Signer(), signers[i]},
+				false,
+			)
+		}
+
+	})
+
+	t.Run("Should register incomplete if only one of the clusters is complete", func(t *testing.T) {
+
+		for i := 0; i < 3; i++ {
+
+			// Construct a valid message and signature
+			msg, _ := hex.DecodeString("deadbeef")
+			validSignature, err := stakingPrivateKeys[i].Sign(msg, collectorVoteHasher)
+			validSignatureString := validSignature.String()[2:]
+			assert.NoError(t, err)
+
+			tx := createTxWithTemplateAndAuthorizer(b, templates.GenerateSubmitVoteScript(env), addresses[i])
+
+			_ = tx.AddArgument(cadence.NewString(validSignatureString))
+			_ = tx.AddArgument(cadence.NewString("deadbeef"))
+
+			signAndSubmit(
+				t, b, tx,
+				[]flow.Address{b.ServiceKey().Address, addresses[i]},
+				[]crypto.Signer{b.ServiceKey().Signer(), signers[i]},
+				false,
+			)
+
+			result := executeScriptAndCheck(t, b, templates.GenerateGetVotingCompletedScript(env), nil)
+			assert.Equal(t, cadence.NewBool(false), result)
+		}
+
+	})
+
+	t.Run("Should register incomplete if a cluster has different vote messages", func(t *testing.T) {
```

Comment:
I think we can add more context on exactly what is being tested here. A cluster having different vote messages is not sufficient on its own to cause the QC aggregation to not be complete. 
```suggestion
	// If a cluster has received votes with weight exceeding the quorum threshold, but those
	// votes are spread across different vote messages such that no set of votes corresponding
	// to any vote message represents weight exceeding the quorum threshold, the cluster voting
	// should be considered incomplete.
	t.Run("Should register incomplete if a cluster has different vote messages", func(t *testing.T) {
```